### PR TITLE
Bluetooth: controller: Refactor to use max macro

### LINKED ISSUE #6230: Bluetooth: controller: refactor to use min/max macro
"Good first issue" to refactor the BLE controller implementation, clean up the use of conditional statements with min/max macros.

Refer to review comment in #6149

### PR DESCRIPTION
**Pull request by DJM0**
Cleaned up the use of conditional statements to use the max macro.

Fixes #6230.

Signed-off-by: David Maitland <hello@davidmaitland.me>

### CODE REVIEW

**Comment by davidmaitland** [comment]
I tested the changes by building and running on a nRF51-PCA10028 board.

I wasn't able to run the sanity tests locally without my laptop crashing (unrelated to my changes), but will try to get that working before committing in the future!

File: subsys/bluetooth/controller/ll_sw/ctrl.c [hunk 1]
```diff
@@ -1132,11 +1132,9 @@ static inline u32_t isr_rx_adv(u8_t devmatch_ok, u8_t devmatch_id,
...
 		conn->hdr.ticks_preempt_to_start =
 			HAL_TICKER_US_TO_TICKS(RADIO_TICKER_PREEMPT_PART_MIN_US);
-		ticks_slot_offset =
-			(conn->hdr.ticks_active_to_start <
-			 conn->hdr.ticks_xtal_to_start) ?
-			conn->hdr.ticks_xtal_to_start :
-			conn->hdr.ticks_active_to_start;
+		ticks_slot_offset = max(
+			conn->hdr.ticks_active_to_start,
+			conn->hdr.ticks_xtal_to_start);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: Have it as below:
```
ticks_slot_offset = max(conn->hdr.ticks_active_to_start,
                        conn->hdr.ticks_xtal_to_start);
```

File: subsys/bluetooth/controller/ll_sw/ctrl.c [hunk 2]
```diff
@@ -4992,11 +4985,9 @@ static void sched_after_mstr_free_slot_get(u8_t user_id,
...
 #if defined(CONFIG_BT_CTLR_XTAL_ADVANCED)
 			if (conn->hdr.ticks_xtal_to_start & XON_BITMASK) {
-				u32_t ticks_prepare_to_start =
-					(conn->hdr.ticks_active_to_start >
-					 conn->hdr.ticks_preempt_to_start) ?
-					conn->hdr.ticks_active_to_start :
-					conn->hdr.ticks_preempt_to_start;
+				u32_t ticks_prepare_to_start = max(
+					conn->hdr.ticks_active_to_start,
+					conn->hdr.ticks_preempt_to_start);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: Here you can have:
```
u32_t ticks_prepare_to_start =
	max(conn->hdr.ticks_active_to_start,
	    conn->hdr.ticks_preempt_to_start);
```

**Review by cvinayak — CHANGES_REQUESTED** [review]
@davidmaitland Alignment style related comments added. Otherwise LGTM.

**Review by carlescufi — APPROVED** [review]
Looks good, thanks!

**Review by lpereira — COMMENTED** [review]
Patch LGTM at a first glance.  It seems that there are two other spots that could get this treatment: ctrl_internal.h (`LL_MEM_RX_POOL_SZ` macro) and ticker.c (`ticker_job_compare_update()`).

**Comment by davidmaitland** [comment]
@lpereira thanks, will have a look.

@cvinayak sorry, didn’t realise I could do that. Should I remove the last commit and do that along with the other bit @lpereira noticed?

**Comment by cvinayak** [comment]
@davidmaitland yes, do all max/min macro updates in one controller commit

**Comment by davidmaitland** [comment]
@cvinayak all squashed and pushed

@lpereira I updated the two spots you noticed, would you mind checking them over?

File: subsys/bluetooth/controller/ll_sw/ctrl_internal.h [hunk 3]
```diff
@@ -327,16 +327,12 @@ struct pdu_data_q_tx {
...
-		      RADIO_LL_LENGTH_OCTETS_RX_MAX) \
-			: \
-		      (PDU_AC_SIZE_MAX + PDU_AC_SIZE_EXTRA))) * \
-			(RADIO_PACKET_COUNT_RX_MAX + 3))
+#define LL_MEM_RX_POOL_SZ (MROUND(\
+			offsetof(struct radio_pdu_node_rx, pdu_data) + (\
+			max((PDU_AC_SIZE_MAX + PDU_AC_SIZE_EXTRA), \
+			    (offsetof(struct pdu_data, lldata) + \
+			    RADIO_LL_LENGTH_OCTETS_RX_MAX))\
+			)) * (RADIO_PACKET_COUNT_RX_MAX + 3))
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ctrl_internal.h]: Alignment, realign with line breaks after operators or operands.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ctrl_internal.h (hunk 3 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ctrl_internal.h]: Below is followed, but I can be wrong too:
```
if ((long test 1) &&
    (long test 2) &&
    (long test 3)) {
  code executed if true;
}
```

**Review by cvinayak — APPROVED** [review]

**Pull request merged**